conduit/src/Data/Conduit/Internal/Conduit.hs

```diff
@@ -720,6 +723,38 @@ connect :: Monad m
         -> m r
 connect = ($$)
 
+-- | Split a conduit into head and tail.
+--
+-- Note that you have to 'sealConduitT' it first.
+--
+-- Since 1.3.3
+uncons :: forall m o conduit. (Monad m, conduit ~ SealedConduitT () o m ())
+       => conduit -> m (Maybe (o, conduit))
```

Comment:
I don't think the equality constraint is a good thing here, it will make the docs harder to understand.